source/Subsystem/EntityComponentSystem.h

```diff
@@ -18,10 +21,32 @@ class EntityComponentSystem
 	Entity* GetEntity(std::string aName);
 	void DestroyEntity(Entity* aEntity);
 
+	template<class T>
+	Component* CreateComponent(Entity* aEntity)
```

Comment:
What happens when you try and add a class that isn't derived from Component or is the Component base class? You might want to consider adding some extra template magic such as std::enable_if & std::is_base_of<Type> to disable functions like these when the interface is improperly used by the coder/user. Only if you want to add the safety layer of course, might still be interesting.
https://www.cplusplus.com/reference/type_traits/enable_if/
https://en.cppreference.com/w/cpp/types/is_base_of